monkey/monkey_island/cc/setup/island_config_options.py

```diff
@@ -1,57 +1,38 @@
 from __future__ import annotations
 
-from types import MappingProxyType as ImmutableMapping
-from typing import Mapping
+from pathlib import Path
+from typing import Annotated
+
+from monkeytypes import InfectionMonkeyBaseModel
+from pydantic import BeforeValidator, Field
 
 from common.utils.file_utils import expand_path
 from monkey_island.cc.server_utils.consts import (
     DEFAULT_CRT_PATH,
     DEFAULT_DATA_DIR,
     DEFAULT_KEY_PATH,
     DEFAULT_LOG_LEVEL,
+    DEFAULT_NODE_PORT,
     DEFAULT_START_MONGO_DB,
 )
 
-_DATA_DIR = "data_dir"
-_SSL_CERT = "ssl_certificate"
-_SSL_CERT_FILE = "ssl_certificate_file"
-_SSL_CERT_KEY = "ssl_certificate_key_file"
-_MONGODB = "mongodb"
-_START_MONGODB = "start_mongodb"
-_LOG_LEVEL = "log_level"
-
-
-class IslandConfigOptions:
-    def __init__(self, config_contents: Mapping[str, Mapping] = ImmutableMapping({})):
-        self.data_dir = DEFAULT_DATA_DIR
-        self.log_level = DEFAULT_LOG_LEVEL
-        self.start_mongodb = DEFAULT_START_MONGO_DB
-        self.crt_path = DEFAULT_CRT_PATH
-        self.key_path = DEFAULT_KEY_PATH
-
-        self._expand_paths()
-
-        self.update(config_contents)
-
-    def update(self, config_contents: Mapping[str, Mapping]):
-        self.data_dir = config_contents.get(_DATA_DIR, self.data_dir)
 
-        self.log_level = config_contents.get(_LOG_LEVEL, self.log_level)
+class MongoDBConfig(InfectionMonkeyBaseModel):
+    start_mongodb: bool = DEFAULT_START_MONGO_DB
 
-        self.start_mongodb = config_contents.get(
-            _MONGODB, {_START_MONGODB: self.start_mongodb}
-        ).get(_START_MONGODB, self.start_mongodb)
 
-        self.crt_path = config_contents.get(_SSL_CERT, {_SSL_CERT_FILE: self.crt_path}).get(
-            _SSL_CERT_FILE, self.crt_path
-        )
-        self.key_path = config_contents.get(_SSL_CERT, {_SSL_CERT_KEY: self.key_path}).get(
-            _SSL_CERT_KEY, self.key_path
-        )
+class SSLCertificatesConfig(InfectionMonkeyBaseModel):
+    ssl_certificate_file: Annotated[
+        Path, Field(default=Path(DEFAULT_CRT_PATH)), BeforeValidator(expand_path)
+    ]
+    ssl_certificate_key_file: Annotated[
+        Path, Field(default=Path(DEFAULT_KEY_PATH)), BeforeValidator(expand_path)
+    ]
```

Comment:
The names of these seem redundant. Calling `config.ssl_certificate.ssl_certificate_file` seems overly verbose. We don't need to say "ssl_certificate" twice someone to know what the variable is for. How about changing them to `certificate_file` and `key_file`?

Reply:
TBH I think these pydantic migration changes are a bit out of scope for an already heavy-hitter PR. Lets do them after the PR as certs have nothing to do with configurable node port

Reply:
Just drop a TODO so we don't forget. In fact, it might be worth dropping the TODO in the changelog. Very hard to miss that way.